dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowProcessor.java

```diff
@@ -122,15 +122,8 @@ public WorkflowProcessor(Contentlet contentlet, User firingUser) {
 				}
 			}
 
-			if (null == action) {
-				if (null != scheme && scheme.isMandatory()) {
-					throw new DotWorkflowException(LanguageUtil
-							.get(user, "message.workflow.error.mandatory.action.type")
-							+ contentlet.getStructure().getName());
-				}
-
+			if (null == action)
```

Comment:
A null action on the processor after the changes we are planning to apply won't be supported, for now add Log.error explaining the processor was called with no action, it will be a good reminder for us.
And don't forget to use curly braces.